filebeat/crawler/crawler.go

```diff
@@ -6,62 +6,70 @@ import (
 
 	"github.com/elastic/beats/filebeat/input/file"
 	"github.com/elastic/beats/filebeat/prospector"
+	"github.com/elastic/beats/filebeat/registrar"
 	"github.com/elastic/beats/libbeat/common"
 	"github.com/elastic/beats/libbeat/logp"
 )
 
 type Crawler struct {
-	prospectors       []*prospector.Prospector
+	prospectors       map[uint64]*prospector.Prospector
 	prospectorConfigs []*common.Config
 	out               prospector.Outlet
 	wg                sync.WaitGroup
+	once              bool
 }
 
-func New(out prospector.Outlet, prospectorConfigs []*common.Config) (*Crawler, error) {
+func New(out prospector.Outlet, prospectorConfigs []*common.Config, once bool) (*Crawler, error) {
 
 	if len(prospectorConfigs) == 0 {
 		return nil, fmt.Errorf("No prospectors defined. You must have at least one prospector defined in the config file.")
 	}
 
 	return &Crawler{
 		out:               out,
+		prospectors:       map[uint64]*prospector.Prospector{},
 		prospectorConfigs: prospectorConfigs,
+		once:              once,
 	}, nil
 }
 
-func (c *Crawler) Start(states file.States, once bool) error {
+func (c *Crawler) Start(r *registrar.Registrar) error {
 
 	logp.Info("Loading Prospectors: %v", len(c.prospectorConfigs))
 
 	// Prospect the globs/paths given on the command line and launch harvesters
 	for _, prospectorConfig := range c.prospectorConfigs {
-
-		prospector, err := prospector.NewProspector(prospectorConfig, states, c.out)
+		err := c.startProspector(prospectorConfig, r.GetStates())
```

Comment:
Yes, and this is intentionally. As long as there are no overlaps in inode/device between the harvesters started of a prospector, this will have no effect. Getting the newest states from the registrar is needed when dynamically starting prospectors. Currently we expect from the user that he makes sure there is no overlap of harvesters. In the future we should introduce a registry for harvesters to detect such cases and warn a user or not start a harvester, if there is already running one with the same state in an other prospector. 

So there is no change in behaviour here (or there shouldn't be). Perhaps I missed something :-(